aten/src/ATen/native/cuda/UpSampleNearest2d.cu

```diff
@@ -160,9 +161,12 @@ static void upsample_nearest2d_out_cuda_template(
       maxThreadsDim[2], std::min<int>(nc, max_threads / block_x / block_y));
   const dim3 block(block_x, block_y, block_z);
 
-  int grid_x = cuda::ATenCeilDiv(output_width, block_x);
-  int grid_y = cuda::ATenCeilDiv(output_height, block_y);
-  int grid_z = cuda::ATenCeilDiv(nc, block_z * 4);
+  int grid_x = std::min<int>(
+      maxGridSize[0], cuda::ATenCeilDiv(output_width, block_x));
+  int grid_y = std::min<int>(
+      maxGridSize[1], cuda::ATenCeilDiv(output_height, block_y));
```

Comment:
Granted, your grid_y and especially grid_x are very unlikely to be more than maxGridSize, but if they ever are, you don't have a loop in the kernel to handle that, so better error out. 
Also, if your grid_x is ever more than maxGridSize[0] (which is 2**31), you'll overflow in your offset computation which uses `int` type.